examples/contrib/DALI/dali_example_utilities.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+
+from ignite.engine import Engine, _prepare_batch
+from ignite.engine import Events
+from ignite.contrib.handlers import ProgressBar
+from ignite.metrics import Accuracy, Loss, RunningAverage
+
+def create_custom_supervised_trainer(model, optimizer, loss_fn, metrics={}, device=None , prepare_batch=None):
```

Comment:
We can reuse `create_supervised_trainer` in the implementation of this `create_custom_supervised_trainer` instead of repeating `_update`

Reply:
Not sure I follow - I did have to make a small change to _update in order to substitute out a custom prepare_batch (that if statement), the default implementation always uses _prepare_batch right?

Would be happy to find a more elegant solution though, as that one feels like a monkey patch

Reply:
Here idea is to do something like that : 
```python
from ignite.engine import create_supervised_trainer


def create_custom_supervised_trainer(model, optimizer, loss_fn, metrics={}, device=None , prepare_batch=None):
    """
    We need to make some changes to the default trainer so we can use running metrics and consume Tensors from DALI
    """

    trainer = create_supervised_trainer(model, optimizer, loss_fn, 
                                                              device=device, non_blocking=non_blocking, 
                                                              prepare_batch=prepare_batch, 
                                                              output_transform=lambda x, y, y_pred, loss: loss.item(), y_pred, y):

    def _metrics_transform(output):
        return output[1], output[2]

    for name, metric in metrics.items():
        metric._output_transform = _metrics_transform
        metric.attach(trainer, name)

    return trainer
```

and even we can avoid writing `_metrics_transform` if `create_supervised_trainer` returns a dictionary `{‘y_pred’: y_pred, ‘y’: y, …}` like described here : https://pytorch.org/ignite/metrics.html